rust/otap-dataflow/crates/otap/src/otap_grpc/client_settings.rs

```diff
@@ -4,12 +4,25 @@
 //! Shared configuration for gRPC-based clients.
 
 use crate::compression::CompressionMethod;
+
+#[cfg(feature = "experimental-tls")]
+use crate::tls_utils;
 use otap_df_config::byte_units;
+#[cfg(feature = "experimental-tls")]
+use otap_df_config::tls::TlsClientConfig;
 use serde::Deserialize;
 use std::time::Duration;
+use thiserror::Error;
 use tonic::codec::CompressionEncoding;
 use tonic::transport::Endpoint;
 
+/// Checks if a URI string starts with "https://" (case-insensitive, per RFC 3986).
+fn is_https_endpoint(uri: &str) -> bool {
+    uri.trim_start()
+        .get(..8)
+        .is_some_and(|s| s.eq_ignore_ascii_case("https://"))
+}
+
```

Comment:
I think this is duplicated, I've seen two of them, moreover I do not want to infer security settings from the URL scheme. (I know this is something OTel SDKs do! This is not how the OTel Collector operates, nor is it part of the OTel declarative configuration as far as I know.)

Reply:
this is no more duplicate now. Thanks,